pkg/types/types.go

```diff
@@ -75,54 +75,115 @@ const (
 	EndpointKindRaw     EndpointKind = "raw"
 )
 
-type CommonData struct {
-	// Node where the event comes from
-	Node string `json:"node,omitempty" column:"node,template:node" columnTags:"kubernetes"`
+type RuntimeName string
 
-	// Pod namespace where the event comes from, or empty for host-level
-	// event
-	Namespace string `json:"namespace,omitempty" column:"namespace,template:namespace" columnTags:"kubernetes"`
+func (r RuntimeName) String() string {
+	return string(r)
+}
 
-	// Pod where the event comes from, or empty for host-level event
-	Pod string `json:"pod,omitempty" column:"pod,template:pod" columnTags:"kubernetes"`
+const (
+	RuntimeNameDocker     RuntimeName = "docker"
+	RuntimeNameContainerd RuntimeName = "containerd"
+	RuntimeNameCrio       RuntimeName = "cri-o"
+	RuntimeNamePodman     RuntimeName = "podman"
+	RuntimeNameUnknown    RuntimeName = "unknown"
+)
 
-	// Container where the event comes from, or empty for host-level or
-	// pod-level event
-	Container string `json:"container,omitempty" column:"container,template:container" columnTags:"kubernetes,runtime"`
+func String2RuntimeName(name string) RuntimeName {
+	switch name {
+	case string(RuntimeNameDocker):
+		return RuntimeNameDocker
+	case string(RuntimeNameContainerd):
+		return RuntimeNameContainerd
+	case string(RuntimeNameCrio):
+		return RuntimeNameCrio
+	case string(RuntimeNamePodman):
+		return RuntimeNamePodman
+	}
+	return RuntimeNameUnknown
+}
+
+type BasicRuntimeMetadata struct {
+	// RuntimeName is the name of the container runtime. It is useful to distinguish
+	// who is the "owner" of each container in a list of containers collected
+	// from multiple runtimes.
+	RuntimeName RuntimeName `json:"runtimeName,omitempty" column:"runtimeName,minWidth:5,maxWidth:12"`
+
+	// ContainerID is the container ContainerID without the container runtime prefix. For
+	// instance, without the "cri-o://" for CRI-O.
+	ContainerID string `json:"containerId,omitempty" column:"containerId,width:13,maxWidth:64"`
```

Comment:
That's a good idea. We don't even need to worry about people parsing it because, in that case, they will use the JSON format where it is divided.

Reply:
I hid the Container ID but not yet the runtime. It is because I'd like to find an intelligent way to automatically add a virtual column (in this case, a column composed by `<runtimeName>://<containerName>) to all the structs that embed `BasicRuntimeMetadata`, instead of modifying the `GetColumns()` of all events to do so. Can I leave it for another PR?

cc @flyth

Reply:
> I'd like to find an intelligent way to automatically add a virtual column (in this case, a column composed by <runtimeName>://<containerName>) to all the structs that embed BasicRuntimeMetadata, instead of modifying the GetColumns() of all events to do so.

Hmm, we could let such a type (like `BasicRuntimeMetadata`) implement an interface that we check for when initializing the columns (e.g. with a function like `InitColumns(columns)`) - where it then could add those virtual columns. But that might increase complexity quite a bit. I think with everything we need for the containerized gadgets in the columns lib, we should sooner or later make a plan with what features we need (in a more dynamic fashion), so we can identify overlaps and hopefully don't increase complexity too much.